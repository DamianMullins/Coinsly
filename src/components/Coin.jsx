import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/Coin.module.scss';
import tick from '../assets/tick.svg';

class Coin extends Component {
  state = {
    loaded: false
  };

  onImageLoaded = () => {
    this.setState({ loaded: true });
  }

  render () {
    const { coin, setOwnedValue } = this.props;

    return (
      <li className={`${styles.coin} ${coin.owned ? styles.coin__owned : ''} ${this.state.loaded ? styles.coin__loaded : 'not-loaded'}`}>
        <label className={styles.label} data-testid="coin-label">
          {coin.owned && <img className={styles.tick} src={tick} alt="" />}

          <img
            className={styles.image}
            src={coin.imageUrl}
            alt=""
            onLoad={this.onImageLoaded} />

          <input
            className={styles.input}
            type="checkbox"
            checked={coin.owned}
            onChange={setOwnedValue}
            value={coin.id}
          />
        </label>
      </li>
    )
  }
}

Coin.propTypes = {
  coin: PropTypes.object.isRequired,
  setOwnedValue: PropTypes.func.isRequired
};

export default Coin;
