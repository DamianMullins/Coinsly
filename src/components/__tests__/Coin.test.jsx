import { describe, it, expect, afterEach } from 'vitest';
import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Coin from '../Coin';

afterEach(cleanup);

it('coin markup is correct', () => {
  // Arrange
  const coin = { id: 2, name: 'Kew Gardens' };
  const component = <Coin coin={coin} setOwnedValue={() => {}} />;

  // Act
  const { container } = render(component);

  // Assert
  expect(container.firstChild).toMatchSnapshot();
});

it('label text content is correct', () => {
  // Arrange
  const coin = { id: 2 };
  const component = <Coin coin={coin} setOwnedValue={() => {}} />;

  // Act
  const { getByTestId } = render(component);

  // Assert
  expect(getByTestId('coin-label').textContent).toBe('');
});
