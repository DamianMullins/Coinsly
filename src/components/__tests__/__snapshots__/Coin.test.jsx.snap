// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`coin markup is correct 1`] = `
<li
  class="_coin_bc0669  not-loaded"
>
  <label
    class="_label_bc0669"
    data-testid="coin-label"
  >
    <img
      alt=""
      class="_image_bc0669"
    />
    <input
      class="_input_bc0669"
      type="checkbox"
      value="2"
    />
  </label>
</li>
`;
